### ISSUE TITLE
LWC Language Server errors in VSCode

### ISSUE THREAD

**Issue by rmccu** [forcedotcom/salesforcedx-vscode#2575]
### Summary

When I open VSCode with a project, the _Output_ panel at the bottom switches to the _LWC Language Server_ and starts showing errors:
```
(node:7697) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'map' of undefined
    at Function.declarationsFromCustomLabels (/Users/username/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-49.11.0/node_modules/@salesforce/lwc-language-server/lib/typing.js:30:54)
    at async Promise.all (index 0)
    at async TypingIndexer.saveCustomLabelTypings (/Users/username/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-49.11.0/node_modules/@salesforce/lwc-language-server/lib/typing-indexer.js:69:31)
(node:7697) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:7697) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[Error - 10:02:44 AM] (node:7697) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'map' of undefined
    at Function.declarationsFromCustomLabels (/Users/username/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-49.11.0/node_modules/@salesforce/lwc-language-server/lib/typing.js:30:54)
    at async Promise.all (index 0)
    at async TypingIndexer.saveCustomLabelTypings (/Users/username/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-49.11.0/node_modules/@salesforce/lwc-language-server/lib/typing-indexer.js:69:31)
[Error - 10:02:44 AM] (node:7697) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
[Error - 10:02:44 AM] (node:7697) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[Error - 10:09:28 AM] Request textDocument/definition failed.
  Message: Request textDocument/definition failed with message: Cannot destructure property 'delimiter' of '(intermediate value)(intermediate value)(intermediate value)' as it is undefined.
  Code: -32603 
```

### Steps To Reproduce:

1.  Launch VSCode
1.  Wait about 15 seconds
1.  Notice errors in _Output_ panel

### Expected result

There should be no errors.

### Actual result

There are errors in the _Output_ panel for _LWC Language Server_.

### Additional information

**VS Code Version**: 1.49.2, LWC Extension v49.11.0

**SFDX CLI Version**: sfdx-cli/7.74.1-32db2396ed darwin-x64 node-v12.18.3

**OS and version**: macOS Catalina (10.15.6)

**Comment by lcampos** [comment]
thanks for opening this issue @rmccu , I haven't been able to replicate this issue on my side yet. Could you share a project that we can use to replicate this issue you are getting ?

**Comment by rmccu** [comment]
@lcampos I emailed you directly.

**Comment by rmccu** [comment]
Ok, I created a github repo with my project: https://github.com/rmccu/mindful-goat

Please let me know if there is additional information needed. Thanks.

**Comment by alechko** [comment]
while I'm not having all of the errors as OP, my VSCode 1.49 constantly throws the following LWC Language Server error (Lightning Web Components extension v49.12.0) all the time, and I'm not working on any LWC components but on Aura components.

```[Error - 7:20:14 PM] Request textDocument/definition failed.
  Message: Request textDocument/definition failed with message: Cannot destructure property 'delimiter' of '(intermediate value)(intermediate value)(intermediate value)' as it is undefined.
  Code: -32603```

I'm on MacOS if that makes any difference.

**Comment by lcampos** [comment]
@alechko you might be running into a different issue since I'm not able to replicate it using the project provided by @rmccu . Do you have a project you can share where this issue is consistently happening ?

**Comment by alechko** [comment]
I'm sorry, I cannot share my project as this is a company project, however I can send a screenshot in which you can see that my lwc folder is empty but the LWC Language Server is throwing an error with id same as the OP's last error.
![Screen Shot 2020-10-07 at 08 55 15](https://user-images.githubusercontent.com/1391846/95292996-44035200-087b-11eb-80cc-3c7a20e9f8a1.png)

**Comment by rmccu** [comment]
@lcampos it looks like a work-around for my issue would be to remove empty custom labels. Is that right?

**Comment by lcampos** [comment]
@rmccu yes, the issue is related to having a custom labels file with no labels in it. The fix is currently under review and hopefully we'll ship it by next week.

@alechko you don't have to share your company's project, any other test project that can be used to consistently replicate the issue would be enough.

**Comment by alechko** [comment]
Alright, I've created a fresh sfdx project, created one aura component, created one lwc component and deleted it, I've recreated the error while pasting something into the aura component, I'll attach a video.

the project: https://github.com/alechko/sfdx-test.git

the video:
![Kapture 2020-10-08 at 8 42 38](https://user-images.githubusercontent.com/1391846/95419905-2c8f9c00-0943-11eb-8283-69dab09e0673.gif)

**Comment by lcampos** [comment]
Thanks for the info @alechko, I was able to replicate it and confirm it's a different issue. I've opened issue #2636 with the info you provided and can follow that one to get updates on the fix.

**Comment by lcampos** [comment]
Closing this since the fix was released in version 50.2.0 of the Salesforce extensions for VSCode. Thanks again for reporting this.
